src/game/Strategic/Strategic_Pathing.cc

```diff
@@ -868,10 +867,18 @@ void RebuildWayPointsForGroupPath(PathSt* const pHeadOfPath, GROUP& g)
 	AddWaypointStrategicIDToPGroup(&g, pNode->uiSectorId);
 
 	// at this point, the final sector in the path must be identical to this group's last waypoint
-	wp = GetFinalWaypoint(&g);
-	AssertMsg( wp, "Path exists, but no waypoints were added!  AM-0" );
-	AssertMsg(pNode->uiSectorId == (UINT32) wp->sSector.AsStrategicIndex(), "Last waypoint differs from final path sector!  AM-0");
+	{
+		auto * const wp = GetFinalWaypoint(&g);
 
+		if (wp == nullptr)
+		{
+			SLOGE("Path exists, but no waypoints were added!  AM-0" );
+		}
+		else if (pNode->uiSectorId != static_cast<UINT32>(wp->sSector.AsStrategicIndex()))
+		{
+			SLOGE("Last waypoint differs from final path sector!  AM-0");
+		}
+	}
```

Comment:
Why the extra block?

Reply:
It makes the code uglier though. Also, if anyone tried to add code that would use wp without ensuring it's not null, coverity would file a complaint anyway.

Reply:
Uglier by two lines and one indentation level? Also, I prefer to not have it when bad code doesn't even compile instead of having to wait until Monday for Coverity to tell me about it (that is, if it finds the problem at all and it doesn't get buried in the other 374 defects).